### ISSUE TITLE
New User challenge

### ISSUE THREAD

**Issue by ASD6789** [fredericojordan/fast-chess#16]
Good day,
I am a new user to this platform and only got to this point as I was googling Chess heat-maps...as I was looking for something similar to the picture shown in the project overview.
I am a recreational chess player and as my work exposes me to data management and visualization - i.e. dashboards, etc. I thought it would be great to have a visual chess board to start learning about the strengths and weaknesses on the board.
Unfortunately, I have not been able to test this - as I have little experience with coding etc. and although the instructions seem simple - download, run .exe, done... I do not seem to be able to find an exe file?
Please let me know as I would be very interested to test this and contribute in ideas or provide feedback from a potential user experience.
As I am also not familiar with GitHub, I do not know how to send a message to the creator - are you also on Discord?
Thank you and regards

**Comment by fredericojordan** [comment]
Hello and welcome!
I am also an amateur/beginner chess player and the heatmap was one of the main ideas behind this project (before I got carried away with the whole idea). 
The installation instructions should work in most platforms, but I've found that it is not always the case.
And sharing compiled binaries is not always so straightforward since the program must be compiled to your specific operational system.

That being said, I have already compiled an `.exe` for Windows on [previous releases](https://github.com/fredericojordan/fast-chess/releases) ([direct link for v1.6.3](https://github.com/fredericojordan/fast-chess/releases/download/v1.6.3/chess-v1.6.3.rar)). Although this is an older version, it should work "out of the box".

Regarding direct messages in Github, currently there is no way of doing that. I am fredericojordan#8702 on Discord, although I'm not very active over there. You may keep opening Issues if you'd like, no problem with me. Any criticism and suggestions are always welcome!

Thanks and have a nice day! 👋

**Comment by fredericojordan** [comment]
And regarding the heatmap, I had the idea of developing a Chrome extension that would enable the heatmap on lichess.org games (since it's the chess website I mostly use).
It would require some time and research on my part, let me know if you also have any interest.

**Comment by ASD6789** [comment]
Thank you for the prompt response! :-D
I mostly play on chess.com - but the idea of a Chrome extension sounds amazing...and quite frankly having the heat-map as an add-on or tool would definitely make me consider switching platforms (i.e. if I had the choice of playing on chess.com without it or lichess.org with it - I would probably use lichess...that is if it works as I imagine...) - it may turn out to be a great learning tool!
Just from the screenshot you posted, immediately it became apparent to my why a3 or h3 are sometimes played early in the game...as there is that red threat...
I would also recommend to have a different color to indicate an equally covered space vs not covered space...say yellow. Hence, if no attackers, then white space...however if white has 1 and black has 1 attack it is light yellow and it increases in yellow intensity as attacks increase.
I am also thinking about a design that maintains the board colors and the heat-map just being additional information, rather than ending up with a completely white board...I'll try to post a picture of what I mean once I have a sketch...
I'm also not very active on Discord - haha...so this thread will be fine for now :-)
But yes - I would definitely be interested in this!

**Comment by ASD6789** [comment]
I just downloaded the .exe for previous release (v1.6.3) and it is such a cool playing experience! :-D

**Comment by fredericojordan** [comment]
> Just from the screenshot you posted, immediately it became apparent to my why a3 or h3 are sometimes played early in the game...as there is that red threat...

Good to know you liked the experience! Getting these insights about a3 and h3 were my motivators as well :)
Be sure to try out edit mode (E hotkey) to analyze a specific position.

> I would also recommend to have a different color to indicate an equally covered space vs not covered space...say yellow. Hence, if no attackers, then white space...however if white has 1 and black has 1 attack it is light yellow and it increases in yellow intensity as attacks increase.

I like your yellow square idea, I'll put it on the roadmap for future releases.
I've also been thinking about how to encode relative piece value into this visualization (for example, defending a square with a pawn isn't the same as defending it with your queen). But I haven't thought of an intuitive way yet, I figure it'd probably get a bit messy. Let me know if you have any ideas on that.

> I am also thinking about a design that maintains the board colors and the heat-map just being additional information, rather than ending up with a completely white board...I'll try to post a picture of what I mean once I have a sketch...

I see your point. Maybe making it have a transparency over the actual board, or encode it in a "ring color", similar to what I did to represent the latest move. I guess both might be valid ideas...

Thanks for the input! 😄

**Comment by ASD6789** [comment]
Yes - I was thinking about how you could represent the piece...e.g. if the King is one of the protectors - sometimes that does not really help...so I was thinking of a value? But I was not sure how this would work...because sometimes having a pawn lead the attack is an advantage, sometimes not...so it would get messy...
I was also thinking more about the yellow color and even that already starts getting messy :-D especially if the board colors are kept... I have a quick possible pic - but I don't think it really works, as it might make people sick or hypnotized if they look at it for too long - hahaha, lol!
![image](https://user-images.githubusercontent.com/69039670/89058561-2dff7f80-d368-11ea-8e6a-7ebe5aaebc40.png)

**Comment by ASD6789** [comment]
So I'll look into some more ideas...Have you seen the Quantum Chess? The initial version had some really cool way of displaying the probabilities - possibly this could be a method as well for the squares?
![image](https://user-images.githubusercontent.com/69039670/89059529-07dadf00-d36a-11ea-9bf3-e05ad0521222.png)
I mean this has now evolved and is very Quantum specific where you can now see the spin as well: https://i.imgur.com/6p2mbsf.gifv
But that is a bit too much physics for my understanding :-D

**Comment by ASD6789** [comment]
This is also an idea...but I think it is a bit underwhelming...it is not as clear as the heat-map...
![image](https://user-images.githubusercontent.com/69039670/89063658-3ad4a100-d371-11ea-9c8f-cd14086e4cbb.png)

**Comment by ASD6789** [comment]
As a starting point, I think the heat-map just needs some grid-lines...
![image](https://user-images.githubusercontent.com/69039670/89066010-5d68b900-d375-11ea-9824-d0748024c003.png)
VS.
![image](https://user-images.githubusercontent.com/69039670/89066079-7f623b80-d375-11ea-8024-c9b1ee33aff6.png)

**Comment by fredericojordan** [comment]
Wow, nice references!
I imagined I probably wasn't the first one to come up with the "chess heatmap" idea, but I had never seen a concrete example!
Thanks for that

I also think the gridline makes complete sense and doesn't look too difficult, I'll try to implement it!
Let me just get my compilation running again and I'll put those new features 😄 
Thanks again!

**Comment by ASD6789** [comment]
Well - when I was looking for chess heat maps, I only found your version and then later on one from http://tlee753.com/chess-visualizer/
I liked the look and feel of your version more, however was missing the yellow element...
The other ones I posted are just images I quickly drew up as a potential way of displaying it - so it is not really a concrete example or working prototype. My friend really liked the one with the red and green bars to display the attacks on a particular square, but I still think the heat map is more visually...so we thought about a toggle between all tree options :-D haha, so either to have the normal board, the attack bars or the heat map.

Are you still considering into turning this into a functional app or chrome extension?

**Comment by fredericojordan** [comment]
Hello!
Although I intend to keep maintaining and improving the app, it always had an experimental nature to it.
That being said, I have no timeline of when the features we discussed are going to be implemented, since I can only work on it on my free time.
On the other hand, the project has a pretty permissive [license](https://github.com/fredericojordan/fast-chess/blob/master/LICENSE), so feel free to copy, adapt and extend anything if you'd like to!

Ps. Regarding the chrome extension, I'd probably have to write entirely new code. So it makes sense to maybe create an entirely new repository exclusively for that idea. But again, no estimates on when that would be, sorry.
